core/templates/dev/head/services/StateRulesStatsService.js

```diff
@@ -0,0 +1,81 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for calculating the statistics of a particular state.
+ */
+
+oppia.factory('StateRulesStatsService', [
+  '$http', '$injector', 'AngularNameService', 'AnswerClassificationService',
+  'ExplorationContextService', 'UrlInterpolationService',
+  function(
+      $http, $injector, AngularNameService, AnswerClassificationService,
+      ExplorationContextService, UrlInterpolationService) {
+    var STATE_RULES_STATS_URL_TEMPLATE =
+      '/createhandler/state_rules_stats/<exploration_id>/<escaped_state_name>';
+
+    return {
+      /**
+       * TODO(brianrodri): Consider moving this into a visualization domain
+       * object.
+       *
+       * @param {state} state
+       * @return {Boolean} whether given state has an implementation for
+       *     displaying the issues overview tab in the State Editor.
+       */
+      stateSupportsIssuesOverview: function(state) {
+        return state.interaction.id === 'TextInput';
+      },
+      /**
+       * Returns a promise which will provide details of the given state's
+       * answer-statistics.
+       */
+      computeStateRulesStats: function(state, explorationId) {
+        if (explorationId === undefined) {
+          explorationId = ExplorationContextService.getExplorationId();
+        }
+        var stateRulesStatsUrl = UrlInterpolationService.interpolateUrl(
+          STATE_RULES_STATS_URL_TEMPLATE, {
+            exploration_id: explorationId,
+            escaped_state_name: state.name
+          }
+        );
+        return $http.get(stateRulesStatsUrl).then(function(response) {
+          var rulesService = $injector.get(
+            AngularNameService.getNameOfInteractionRulesService(
+              state.interaction.id));
+          var stateRulesStats = {
```

Comment:
I think this ought to be a frontend domain object, too.

Reply:
Acknowledged, I'll do that in a separate PR.